src/nvim/ex_docmd.c

```diff
@@ -568,14 +568,19 @@ int do_cmdline(char_u *cmdline, LineGetter fgetline,
        * from a script or when being called recursive (e.g. for ":e
        * +command file").
        */
-      if (!(flags & DOCMD_NOWAIT) && !recursive) {
+      bool isPreview = (State & CMDPREVIEW) == 0 ? false : true;
+      if (!((flags & DOCMD_NOWAIT) || recursive) || isPreview) {
         msg_didout_before_start = msg_didout;
-        msg_didany = FALSE;         /* no output yet */
+        msg_didany = false;         // no output yet
         msg_start();
-        msg_scroll = TRUE;          /* put messages below each other */
-        ++no_wait_return;           /* don't wait for return until finished */
-        ++RedrawingDisabled;
-        did_inc = TRUE;
+        msg_scroll = true;          // put messages below each other
+        if (!isPreview) {
+          // For execution of a preview command we have to wait for a return
+          // key, but we don't want the message for it pop out
+          no_wait_return++;           // don't wait for return until finished
```

Comment:
> but we don't want the message for it pop out

If we _don't_ want to wait, why `no_wait_return++` for `!isPreview` ?

Reply:
Ops, may be wrong msg, I think I was thinking some otherthing, I also change it